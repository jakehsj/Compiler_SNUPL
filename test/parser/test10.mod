//
// test10
//
// parser test:
// - arrays
//
module test10;

var a: integer[100][5];
    i: integer;

begin
  i := 0;
  while (i < 100) do
    a[i] := i
  end
end test10.
