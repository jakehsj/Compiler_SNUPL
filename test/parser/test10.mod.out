parsing '../test/parser/test10.mod'...
successfully parsed.
  AST:
    CAstScope: 'test10'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <NULL>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<integer>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to <char>>) --> <NULL>           ]
          [ @a        <integer>           a           ]
          [ @i        <integer>           i           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        := <integer>
          [ @i        <integer>           i ] <integer>
          0 <integer>
        while cond
          < <integer>
            [ @i        <integer>           i ] <integer>
            100 <integer>
        while-body
          := <INVALID>
            [ @a        <integer>           a ] <INVALID>
              [ @i        <integer>           i ] <integer>
            [ @i        <integer>           i ] <integer>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -o../test/parser/test10.mod.ast.pdf ../test/parser/test10.mod.ast.dot


Done.
